File: packages/ui-swiftui/Sources/KinetixUI/Sheet.swift
//
// Sheet.swift — KinetixSheet family.
//
// Mirrors packages/ui/src/components/sheet.tsx. The React `side`
// (top/bottom/left/right) collapses to **bottom only** here — the
// idiomatic mobile pattern and the only edge with a clean single
// primitive, the same deliberate scope cut the Compose port made
// (`ModalBottomSheet`). Place it in an `.overlay { }` or top-level
// `ZStack` and drive it with an `isPresented` binding; wrap the state
// change in `withAnimation` for the slide-up.
//
// Header / Title / Description / Footer are the KinetixDialog* parts
// (identical on the web).
//

import SwiftUI

public struct KinetixSheet<Content: View>: View {
    @Environment(\.kinetixColors) private var colors

    @Binding private var isPresented: Bool
    private let content: Content

    public init(isPresented: Binding<Bool>, @ViewBuilder content: () -> Content) {
        self._isPresented = isPresented
        self.content = content()
    }

    public var body: some View {
        if isPresented {
            ZStack(alignment: .bottom) {
                Color.black.opacity(0.4)
                    .ignoresSafeArea()
                    .onTapGesture { isPresented = false }

                VStack(alignment: .leading, spacing: 16) {
                    Capsule()
                        .fill(colors.border)
                        .frame(width: 36, height: 4)
                        .frame(maxWidth: .infinity) // grabber
                    content
                }
                .padding(24) // p-6
                .frame(maxWidth: .infinity, alignment: .leading)
                .background(
                    colors.background,
                    in: RoundedRectangle(cornerRadius: 16, style: .continuous) // radius/xl, top corners visually
                )
                .overlay(alignment: .top) {
                    RoundedRectangle(cornerRadius: 16, style: .continuous)
                        .strokeBorder(colors.border, lineWidth: 1)
                }
                .shadow(color: .black.opacity(0.2), radius: 16, y: -4)
                .ignoresSafeArea(edges: .bottom)
                .transition(.move(edge: .bottom))
            }
        }
    }
}

public typealias KinetixSheetHeader<Content: View> = KinetixDialogHeader<Content>
public typealias KinetixSheetFooter<Content: View> = KinetixDialogFooter<Content>
public typealias KinetixSheetTitle = KinetixDialogTitle
public typealias KinetixSheetDescription = KinetixDialogDescription

File: packages/ui-flutter/lib/src/sheet.dart
import 'package:flutter/material.dart';

import 'dialog.dart';
import 'theme.dart';

/// Mirrors `packages/ui/src/components/sheet.tsx`. Collapsed to **bottom
/// only** — the idiomatic mobile pattern and the only edge with a clean
/// primitive, the same scope cut the other ports made. Place it in a
/// top-level `Stack`, driven by `visible`.
///
/// Header / Title / Description / Footer are the KinetixDialog* parts.
class KinetixSheet extends StatelessWidget {
  const KinetixSheet({
    super.key,
    required this.visible,
    required this.onDismiss,
    required this.child,
  });

  final bool visible;
  final VoidCallback onDismiss;
  final Widget child;

  @override
  Widget build(BuildContext context) {
    if (!visible) return const SizedBox.shrink();
    final c = KinetixTheme.of(context);

    return Stack(
      children: [
        Positioned.fill(
          child: GestureDetector(
            onTap: onDismiss,
            child: const ColoredBox(color: Color(0x66000000)),
          ),
        ),
        Align(
          alignment: Alignment.bottomCenter,
          child: Material(
            color: c.background,
            borderRadius: const BorderRadius.vertical(top: Radius.circular(16)),
            elevation: 8,
            child: Container(
              width: double.infinity,
              padding: const EdgeInsets.all(24), // p-6
              child: Column(
                mainAxisSize: MainAxisSize.min,
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Center(
                    child: Container(
                      width: 36,
                      height: 4,
                      decoration: BoxDecoration(
                        color: c.border,
                        borderRadius: BorderRadius.circular(9999),
                      ),
                    ),
                  ),
                  const SizedBox(height: 16),
                  DefaultTextStyle.merge(
                    style: TextStyle(color: c.foreground, fontSize: 14),
                    child: child,
                  ),
                ],
              ),
            ),
          ),
        ),
      ],
    );
  }
}

// The Dialog header parts double as the Sheet header parts (identical on the web).
typedef KinetixSheetHeader = KinetixDialogHeader;
typedef KinetixSheetFooter = KinetixDialogFooter;
typedef KinetixSheetTitle = KinetixDialogTitle;
typedef KinetixSheetDescription = KinetixDialogDescription;
